src/flagpole/conditions.py

```diff
@@ -156,101 +152,83 @@ def _operator_match(self, condition_property: Any, segment_name: str):
-    | list[StrictStr]
-)
+EqualsOperatorValueTypes = int | float | str | bool | list[int] | list[float] | list[str]
 
 
 class EqualsCondition(ConditionBase):
-    operator: Literal[ConditionOperatorKind.EQUALS] = ConditionOperatorKind.EQUALS
     value: EqualsOperatorValueTypes
-    strict_validation: bool = Field(
-        description="Whether the condition should enable strict validation, raising an exception if the evaluation context property is missing",
-        default=False,
-    )
-    """Whether the condition should enable strict validation, raising an exception if the evaluation context property is missing"""
+    operator: str = dataclasses.field(default="equals")
 
     def _operator_match(self, condition_property: Any, segment_name: str):
         return self._evaluate_equals(
             condition_property=condition_property,
             segment_name=segment_name,
-            strict_validation=self.strict_validation,
         )
 
 
 class NotEqualsCondition(ConditionBase):
-    operator: Literal[ConditionOperatorKind.NOT_EQUALS] = ConditionOperatorKind.NOT_EQUALS
     value: EqualsOperatorValueTypes
-    strict_validation: bool = Field(
-        description="Whether the condition should enable strict validation, raising an exception if the evaluation context property is missing",
-        default=False,
-    )
-    """Whether the condition should enable strict validation, raising an exception if the evaluation context property is missing"""
+    operator: str = dataclasses.field(default="not_equals")
 
     def _operator_match(self, condition_property: Any, segment_name: str):
         return not self._evaluate_equals(
             condition_property=condition_property,
             segment_name=segment_name,
-            strict_validation=self.strict_validation,
         )
 
 
-# We have to group and annotate all the different subclasses of Operator
-# in order for Pydantic to be able to discern between the different types
-# when parsing a dict or JSON.
-AvailableConditions = Annotated[
-    InCondition
-    | NotInCondition
-    | ContainsCondition
-    | NotContainsCondition
-    | EqualsCondition
-    | NotEqualsCondition,
-    Field(discriminator="operator"),
-]
+OPERATOR_LOOKUP: Mapping[ConditionOperatorKind, type[ConditionBase]] = {
+    ConditionOperatorKind.IN: InCondition,
+    ConditionOperatorKind.NOT_IN: NotInCondition,
+    ConditionOperatorKind.CONTAINS: ContainsCondition,
+    ConditionOperatorKind.NOT_CONTAINS: NotContainsCondition,
+    ConditionOperatorKind.EQUALS: EqualsCondition,
+    ConditionOperatorKind.NOT_EQUALS: NotEqualsCondition,
+}
 
 
-class Segment(BaseModel):
-    name: constr(min_length=1) = Field(  # type:ignore[valid-type]
-        description="A brief description or identifier for the segment"
+def condition_from_dict(data: Mapping[str, Any]) -> ConditionBase:
+    operator_kind = ConditionOperatorKind(data.get("operator", "invalid"))
+    condition_cls = OPERATOR_LOOKUP[operator_kind]
```

Comment:
Do we want to raise explicitly here if we get an invalid operator type? Otherwise, won't this be an attribute error on NoneType?